pkg/api/owners.go

```diff
@@ -0,0 +1,779 @@
+	return f.ClaimSimple(FieldMemReservation, plugin)
+}
+
+func (f *FieldOwners) ClaimMemSwapLimit(plugin string) error {
+	return f.ClaimSimple(FieldMemSwapLimit, plugin)
+}
+
+func (f *FieldOwners) ClaimMemKernelLimit(plugin string) error {
+	return f.ClaimSimple(FieldMemKernelLimit, plugin)
+}
+
+func (f *FieldOwners) ClaimMemTcpLimit(plugin string) error {
+	return f.ClaimSimple(FieldMemTcpLimit, plugin)
+}
+
+func (f *FieldOwners) ClaimMemSwappiness(plugin string) error {
+	return f.ClaimSimple(FieldMemSwappiness, plugin)
+}
+
+func (f *FieldOwners) ClaimMemDisableOomKiller(plugin string) error {
+	return f.ClaimSimple(FieldMemDisableOomKiller, plugin)
+}
+
+func (f *FieldOwners) ClaimMemUseHierarchy(plugin string) error {
+	return f.ClaimSimple(FieldMemUseHierarchy, plugin)
+}
+
+func (f *FieldOwners) ClaimCpuShares(plugin string) error {
+	return f.ClaimSimple(FieldCpuShares, plugin)
+}
+
+func (f *FieldOwners) ClaimCpuQuota(plugin string) error {
+	return f.ClaimSimple(FieldCpuQuota, plugin)
+}
+
+func (f *FieldOwners) ClaimCpuPeriod(plugin string) error {
+	return f.ClaimSimple(FieldCpuPeriod, plugin)
+}
+
+func (f *FieldOwners) ClaimCpuRealtimeRuntime(plugin string) error {
+	return f.ClaimSimple(FieldCpuRealtimeRuntime, plugin)
+}
+
+func (f *FieldOwners) ClaimCpuRealtimePeriod(plugin string) error {
+	return f.ClaimSimple(FieldCpuRealtimePeriod, plugin)
+}
+
+func (f *FieldOwners) ClaimCpusetCpus(plugin string) error {
+	return f.ClaimSimple(FieldCpusetCpus, plugin)
+}
+
+func (f *FieldOwners) ClaimCpusetMems(plugin string) error {
+	return f.ClaimSimple(FieldCpusetMems, plugin)
+}
+
+func (f *FieldOwners) ClaimPidsLimit(plugin string) error {
+	return f.ClaimSimple(FieldPidsLimit, plugin)
+}
+
+func (f *FieldOwners) ClaimHugepageLimit(size, plugin string) error {
+	return f.ClaimCompound(FieldHugepageLimits, size, plugin)
+}
+
+func (f *FieldOwners) ClaimBlockioClass(plugin string) error {
+	return f.ClaimSimple(FieldBlockioClass, plugin)
+}
+
+func (f *FieldOwners) ClaimRdtClass(plugin string) error {
+	return f.ClaimSimple(FieldRdtClass, plugin)
+}
+
+func (f *FieldOwners) ClaimCgroupsUnified(key, plugin string) error {
+	return f.ClaimCompound(FieldCgroupsUnified, key, plugin)
+}
+
+func (f *FieldOwners) ClaimCgroupsPath(plugin string) error {
+	return f.ClaimSimple(FieldCgroupsPath, plugin)
+}
+
+func (f *FieldOwners) ClaimOomScoreAdj(plugin string) error {
+	return f.ClaimSimple(FieldOomScoreAdj, plugin)
+}
+
+func (f *FieldOwners) ClaimRlimit(typ, plugin string) error {
+	return f.ClaimCompound(FieldRlimits, typ, plugin)
+}
+
+func (f *FieldOwners) ClearCompound(field int32, key, plugin string) {
+	m, ok := f.Compound[field]
+	if !ok {
+		m = NewCompoundFieldOwners()
+		f.Compound[field] = m
+	}
+
+	m.Owners[key] = MarkForRemoval(plugin)
+}
+
+func (f *FieldOwners) ClearSimple(field int32, plugin string) {
+	f.Simple[field] = MarkForRemoval(plugin)
+}
+
+func (f *FieldOwners) ClearAnnotation(key, plugin string) {
```

Comment:
These `Clear` functions probably deserve a bit of a comment explaining the situations in which they should be called.  It's not immediately obvious why we'd clear ownership for an existing field.

Reply:
Actually none of these functions are called by plugins directly. Both the claiming and the clearing functions are used internally in pkg/adaptation to track container adjustments and updates requested by plugins, and to detect conflicts between requests. Added clarifying comments for this.